staging/src/k8s.io/cli-runtime/pkg/resource/mapper.go

```diff
@@ -64,6 +65,10 @@ func (m *mapper) infoForData(data []byte, source string) (*Info, error) {
 		}
 		mapping, err := restMapper.RESTMapping(gvk.GroupKind(), gvk.Version)
 		if err != nil {
+			if _, ok := err.(*meta.NoKindMatchError); ok {
+				return nil, fmt.Errorf("resource mapping not found for name: %q namespace: %q from %q: %v\nensure CRDs are installed first",
```

Comment:
the namespace is not a parameter of the function that errors, does it make sense to include it? Also I find it odd to refer to CRDs while we don't really know if it's a CRD in the first place?

Reply:
I think the namespace makes sense because it's gotten from decoding the object.

This is to provide a hint to the user since we don't want to orchestrate applying CRDs first via kubectl. Do you think it should say something else? "hint: if working with custom resources ensure the CRD is installed first"?